src/libmain.c

```diff
@@ -765,16 +769,20 @@ static gint setup_config_dir(void)
 	return mkdir_result;
 }
 
-/* Signal handling removed since main_quit() uses functions that are
- * illegal in signal handlers
-static void signal_cb(gint sig)
+
+#ifdef G_OS_UNIX
+static gboolean signal_cb(gpointer user_data)
 {
+	gint sig = GPOINTER_TO_INT(user_data);
 	if (sig == SIGTERM)
```

Comment:
what about having a signal-specific callback instead of checking this?  `g_unix_signal_add()` can only get one signum at a time IIUC, so there doesn't seem to be any reason to reuse the same handler when the body will be totally different each time.

So I'd suggest renaming to something like `sigterm_handler()` or something, and remove the signum check (and passing it altogether).